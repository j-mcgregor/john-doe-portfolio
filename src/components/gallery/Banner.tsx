import React from 'react'

const Banner: React.FC = ({ children }) => {
    return <div className="text-center pb3 pt5">{children}</div>
}

export default Banner
