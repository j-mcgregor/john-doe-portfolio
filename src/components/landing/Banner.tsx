import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
    faReact,
    faJs,
    faSass,
    faDocker,
    faGithubAlt,
} from '@fortawesome/free-brands-svg-icons'
import styled from 'styled-components'
import { Breakpoints } from '../../types/enums'

const StyledBanner = styled.div`
    svg {
        font-size: 3.5em;
    }

    @media screen and (max-width: ${Breakpoints.md}px) {
        svg {
            font-size: 3em;
            margin-left: 0.8em;
            margin-right: 0.8em;
        }
    }
    @media screen and (max-width: ${Breakpoints.sm}px) {
        svg {
            font-size: 2em;
            margin-left: 0.5em;
            margin-right: 0.5em;
        }
    }
`

const Banner = () => {
    return (
        <StyledBanner className="p3">
            <div className="container flex flex-row flex-center">
                <FontAwesomeIcon icon={faJs} className="mx1" />
                <FontAwesomeIcon icon={faReact} className="mx1" />
                <FontAwesomeIcon icon={faSass} className="mx1" />
                <FontAwesomeIcon icon={faDocker} className="mx1" />
                <FontAwesomeIcon icon={faGithubAlt} className="mx1" />
            </div>
        </StyledBanner>
    )
}

export default Banner
