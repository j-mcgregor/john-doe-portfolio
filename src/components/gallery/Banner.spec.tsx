import React from 'react'
import { render } from '@testing-library/react'
import Banner from './Banner'

describe('Banner', () => {
    it('should render', () => {
        const { baseElement } = render(<Banner />)
        expect(baseElement).toMatchSnapshot()
    })
})
