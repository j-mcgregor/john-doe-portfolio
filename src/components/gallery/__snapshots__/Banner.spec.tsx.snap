// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render 1`] = `
<body>
  <div>
    <div
      class="text-center pb3 pt5 px2"
    />
  </div>
</body>
`;
